components/feature/syncedtabs/src/main/java/mozilla/components/feature/syncedtabs/internal/SyncedTabsPresenter.kt

```diff
@@ -0,0 +1,116 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.syncedtabs.internal
+
+import androidx.annotation.VisibleForTesting
+import androidx.lifecycle.LifecycleOwner
+import mozilla.components.concept.sync.AccountObserver
+import mozilla.components.concept.sync.AuthType
+import mozilla.components.concept.sync.OAuthAccount
+import mozilla.components.feature.syncedtabs.view.SyncedTabsView
+import mozilla.components.feature.syncedtabs.view.SyncedTabsView.ErrorType
+import mozilla.components.service.fxa.SyncEngine
+import mozilla.components.service.fxa.manager.FxaAccountManager
+import mozilla.components.service.fxa.manager.SyncEnginesStorage
+import mozilla.components.service.fxa.sync.SyncStatusObserver
+
+/**
+ * The tricky part in this class is to handle all possible Sync+FxA states:
+ *
+ * - No Sync account
+ * - Connected to FxA but not Sync (impossible state on mobile at the moment).
+ * - Connected to Sync, but needs reconnection.
+ * - Connected to Sync, but tabs syncing disabled.
+ * - Connected to Sync, but tabs haven't been synced yet (they stay in memory after the first sync).
+ * - Connected to Sync, but only one device in the account (us), so no other tab to show.
+ * - Connected to Sync.
+ *
+ */
+internal class SyncedTabsPresenter(
+    private val controller: SyncedTabsController,
+    private val accountManager: FxaAccountManager,
+    private val view: SyncedTabsView,
+    private val lifecycleOwner: LifecycleOwner
+) {
+
+    @VisibleForTesting
+    internal val eventObserver = SyncedTabsSyncObserver(view, controller)
+    @VisibleForTesting
+    internal val accountObserver = SyncedTabsAccountObserver(view, controller)
+
+    fun start() {
+
+        // No account setup
+        if (accountManager.accountProfile() == null) {
```

Comment:
The account manager should grow a `hasAccount` method (we did this for [iOS](https://searchfox.org/mozilla-mobile/rev/acbd008811cc7e733be6dcf568a26405a90c600b/application-services/components/fxa-client/ios/FxAClient/FxAccountManager.swift#70)) cc @grigoryk